### ISSUE TITLE
Compilation fails under VS 2017 15.8 preview 3

### ISSUE THREAD

**Issue by reupen** [reupen/columns_ui#127]
With the following errors:

```
2>f:\foobar2000\columns_ui\foo_ui_columns\drop_down_list_toolbar.h(103,21): error C3861:  'name': identifier not found
2>f:\foobar2000\columns_ui\foo_ui_columns\drop_down_list_toolbar.h(103,21): error C3861:     for (auto& [id, name] : m_items) {
2>f:\foobar2000\columns_ui\foo_ui_columns\drop_down_list_toolbar.h(103,21): error C3861:                     ^ (compiling source file dsp_preset.cpp)
2>f:\foobar2000\columns_ui\foo_ui_columns\drop_down_list_toolbar.h(103,17): error C3861:  'id': identifier not found
2>f:\foobar2000\columns_ui\foo_ui_columns\drop_down_list_toolbar.h(103,17): error C3861:     for (auto& [id, name] : m_items) {
2>f:\foobar2000\columns_ui\foo_ui_columns\drop_down_list_toolbar.h(103,17): error C3861:                 ^ (compiling source file .\order_dropdown.cpp)
```

It looks like a compiler bug to me, so it's been reported at:

https://developercommunity.visualstudio.com/content/problem/282516/structured-bindings-plus-templates-compiler-error.html

**Comment by reupen** [comment]
According to [the problem report](https://developercommunity.visualstudio.com/content/problem/282516/structured-bindings-plus-templates-compiler-error.html), it's been fixed for Visual Studio 2017 15.8 Preview 5.

In the meantime, 15.7 continues to work.

**Comment by reupen** [comment]
Both reported problems are indeed fixed in Visual Studio 2017 15.8 preview 5.
